Cannot build with LLVM 11

Currently, we can only get Windows builds and tests working for remill and anvill with LLVM 11, so supporting LLVM 11 in McSema would be ideal to integrate Windows into McSema's CI.

I get the following errors when trying to build with LLVM 11
```
FAILED: CMakeFiles/mcsema-lift-11.0.dir/mcsema/BC/Function.cpp.o
/home/ekilmer/.local/bin/ccache /usr/bin/clang++ -DGFLAGS_DLL_DECLARE_FLAG="" -DGFLAGS_DLL_DEFINE_FLAG="" -DGFLAGS_IS_A_DLL=0 -DGOOGLE_GLOG_DLL_DECL="" -DGOOGLE_PROTOBUF_NO_RTTI -DNDEBUG -DREMILL_BUILD_SEMANTICS_DIR_AARCH64=\"/home/ekilmer/github/cxx-common/remill/build/lib/Arch/AArch64/Runtime\" -DREMILL_BUILD_SEMANTICS_DIR_SPARC32=\"/home/ekilmer/github/cxx-common/remill/build/lib/Arch/SPARC32/Runtime\" -DREMILL_BUILD_SEMANTICS_DIR_SPARC64=\"/home/ekilmer/github/cxx-common/remill/build/lib/Arch/SPARC64/Runtime\" -DREMILL_BUILD_SEMANTICS_DIR_X86=\"/home/ekilmer/github/cxx-common/remill/build/lib/Arch/X86/Runtime\" -DREMILL_INSTALL_SEMANTICS_DIR=\"/home/ekilmer/github/cxx-common/remill/build/install/share/remill/11/semantics\" -D_GNU_SOURCE -D_DEBUG -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -isystem . -isystem ../ -isystem /home/ekilmer/github/cxx-common/remill/build/install/include -isystem /home/ekilmer/github/cxx-common/vcpkg/installed/x64-linux-rel/include -isystem /home/ekilmer/github/cxx-common/vcpkg/installed/x64-linux-rel/include/xed -isystem /home/ekilmer/github/cxx-common/anvill/build/install/include -O2 -g -DNDEBUG -Wconversion -pedantic -Wno-unreachable-code-return -Wall -Wextra -Wno-unused-parameter -Wno-c++98-compat -Wno-nested-anon-types -Wno-extended-offsetof -Wno-variadic-macros -Wno-return-type-c-linkage -Wno-c99-extensions -Wno-ignored-attributes -Wno-unused-local-typedef -Wno-unknown-pragmas -Wno-unknown-warning-option -fPIC -fno-omit-frame-pointer -fvisibility-inlines-hidden -fno-asynchronous-unwind-tables -gdwarf-2 -g3 -O2 -std=c++17 -MD -MT CMakeFiles/mcsema-lift-11.0.dir/mcsema/BC/Function.cpp.o -MF CMakeFiles/mcsema-lift-11.0.dir/mcsema/BC/Function.cpp.o.d -o CMakeFiles/mcsema-lift-11.0.dir/mcsema/BC/Function.cpp.o -c ../mcsema/BC/Function.cpp
../mcsema/BC/Function.cpp:28:10: fatal error: 'llvm/IR/CallSite.h' file not found
#include <llvm/IR/CallSite.h>
         ^~~~~~~~~~~~~~~~~~~~
1 error generated.
[24/31] Building CXX object CMakeFiles/mcsema-lift-11.0.dir/tools/mcsema_lift/Lift.cpp.o
FAILED: CMakeFiles/mcsema-lift-11.0.dir/tools/mcsema_lift/Lift.cpp.o
/home/ekilmer/.local/bin/ccache /usr/bin/clang++ -DGFLAGS_DLL_DECLARE_FLAG="" -DGFLAGS_DLL_DEFINE_FLAG="" -DGFLAGS_IS_A_DLL=0 -DGOOGLE_GLOG_DLL_DECL="" -DGOOGLE_PROTOBUF_NO_RTTI -DNDEBUG -DREMILL_BUILD_SEMANTICS_DIR_AARCH64=\"/home/ekilmer/github/cxx-common/remill/build/lib/Arch/AArch64/Runtime\" -DREMILL_BUILD_SEMANTICS_DIR_SPARC32=\"/home/ekilmer/github/cxx-common/remill/build/lib/Arch/SPARC32/Runtime\" -DREMILL_BUILD_SEMANTICS_DIR_SPARC64=\"/home/ekilmer/github/cxx-common/remill/build/lib/Arch/SPARC64/Runtime\" -DREMILL_BUILD_SEMANTICS_DIR_X86=\"/home/ekilmer/github/cxx-common/remill/build/lib/Arch/X86/Runtime\" -DREMILL_INSTALL_SEMANTICS_DIR=\"/home/ekilmer/github/cxx-common/remill/build/install/share/remill/11/semantics\" -D_GNU_SOURCE -D_DEBUG -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -isystem . -isystem ../ -isystem /home/ekilmer/github/cxx-common/remill/build/install/include -isystem /home/ekilmer/github/cxx-common/vcpkg/installed/x64-linux-rel/include -isystem /home/ekilmer/github/cxx-common/vcpkg/installed/x64-linux-rel/include/xed -isystem /home/ekilmer/github/cxx-common/anvill/build/install/include -O2 -g -DNDEBUG -Wconversion -pedantic -Wno-unreachable-code-return -Wall -Wextra -Wno-unused-parameter -Wno-c++98-compat -Wno-nested-anon-types -Wno-extended-offsetof -Wno-variadic-macros -Wno-return-type-c-linkage -Wno-c99-extensions -Wno-ignored-attributes -Wno-unused-local-typedef -Wno-unknown-pragmas -Wno-unknown-warning-option -fPIC -fno-omit-frame-pointer -fvisibility-inlines-hidden -fno-asynchronous-unwind-tables -gdwarf-2 -g3 -O2 -std=c++17 -MD -MT CMakeFiles/mcsema-lift-11.0.dir/tools/mcsema_lift/Lift.cpp.o -MF CMakeFiles/mcsema-lift-11.0.dir/tools/mcsema_lift/Lift.cpp.o.d -o CMakeFiles/mcsema-lift-11.0.dir/tools/mcsema_lift/Lift.cpp.o -c ../tools/mcsema_lift/Lift.cpp
../tools/mcsema_lift/Lift.cpp:308:28: error: no viable conversion from 'llvm::StringRef' to 'const std::string' (aka 'const basic_string<char>')
        CloneFunction(*fn, alias.getName());
                           ^~~~~~~~~~~~~~~
/usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/basic_string.h:448:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'const std::__cxx11::basic_string<char> &' for 1st argument
      basic_string(const basic_string& __str)
      ^
/usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/basic_string.h:525:7: note: candidate constructor template not viable: no known conversion from 'llvm::StringRef' to 'const char *' for 1st argument
      basic_string(const _CharT* __s, const _Alloc& __a = _Alloc())
      ^
/usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/basic_string.h:552:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'std::__cxx11::basic_string<char> &&' for 1st argument
      basic_string(basic_string&& __str) noexcept
      ^
/usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/basic_string.h:579:7: note: candidate constructor not viable: no known conversion from 'llvm::StringRef' to 'initializer_list<char>' for 1st argument
      basic_string(initializer_list<_CharT> __l, const _Alloc& __a = _Alloc())
      ^
/usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/basic_string.h:440:7: note: explicit constructor is not a candidate
      basic_string(const _Alloc& __a) _GLIBCXX_NOEXCEPT
      ^
/usr/lib/gcc/x86_64-redhat-linux/10/../../../../include/c++/10/bits/basic_string.h:650:2: note: explicit constructor is not a candidate
        basic_string(const _Tp& __t, const _Alloc& __a = _Alloc())
        ^
/home/ekilmer/github/cxx-common/vcpkg/installed/x64-linux-rel/include/llvm/ADT/StringRef.h:275:14: note: explicit conversion function is not a candidate
    explicit operator std::string() const { return str(); }
             ^
/home/ekilmer/github/cxx-common/vcpkg/installed/x64-linux-rel/include/llvm/ADT/StringRef.h:278:5: note: candidate function
    operator std::string_view() const {
    ^
../tools/mcsema_lift/Lift.cpp:252:63: note: passing argument to parameter 'name' here
  void CloneFunction(llvm::Function &func, const std::string &name = "") {
                                                              ^
1 error generated.
[25/31] Building CXX object CMakeFiles/mcsema-lift-11.0.dir/mcsema/BC/Optimize.cpp.o
FAILED: CMakeFiles/mcsema-lift-11.0.dir/mcsema/BC/Optimize.cpp.o
/home/ekilmer/.local/bin/ccache /usr/bin/clang++ -DGFLAGS_DLL_DECLARE_FLAG="" -DGFLAGS_DLL_DEFINE_FLAG="" -DGFLAGS_IS_A_DLL=0 -DGOOGLE_GLOG_DLL_DECL="" -DGOOGLE_PROTOBUF_NO_RTTI -DNDEBUG -DREMILL_BUILD_SEMANTICS_DIR_AARCH64=\"/home/ekilmer/github/cxx-common/remill/build/lib/Arch/AArch64/Runtime\" -DREMILL_BUILD_SEMANTICS_DIR_SPARC32=\"/home/ekilmer/github/cxx-common/remill/build/lib/Arch/SPARC32/Runtime\" -DREMILL_BUILD_SEMANTICS_DIR_SPARC64=\"/home/ekilmer/github/cxx-common/remill/build/lib/Arch/SPARC64/Runtime\" -DREMILL_BUILD_SEMANTICS_DIR_X86=\"/home/ekilmer/github/cxx-common/remill/build/lib/Arch/X86/Runtime\" -DREMILL_INSTALL_SEMANTICS_DIR=\"/home/ekilmer/github/cxx-common/remill/build/install/share/remill/11/semantics\" -D_GNU_SOURCE -D_DEBUG -D__STDC_CONSTANT_MACROS -D__STDC_FORMAT_MACROS -D__STDC_LIMIT_MACROS -isystem . -isystem ../ -isystem /home/ekilmer/github/cxx-common/remill/build/install/include -isystem /home/ekilmer/github/cxx-common/vcpkg/installed/x64-linux-rel/include -isystem /home/ekilmer/github/cxx-common/vcpkg/installed/x64-linux-rel/include/xed -isystem /home/ekilmer/github/cxx-common/anvill/build/install/include -O2 -g -DNDEBUG -Wconversion -pedantic -Wno-unreachable-code-return -Wall -Wextra -Wno-unused-parameter -Wno-c++98-compat -Wno-nested-anon-types -Wno-extended-offsetof -Wno-variadic-macros -Wno-return-type-c-linkage -Wno-c99-extensions -Wno-ignored-attributes -Wno-unused-local-typedef -Wno-unknown-pragmas -Wno-unknown-warning-option -fPIC -fno-omit-frame-pointer -fvisibility-inlines-hidden -fno-asynchronous-unwind-tables -gdwarf-2 -g3 -O2 -std=c++17 -MD -MT CMakeFiles/mcsema-lift-11.0.dir/mcsema/BC/Optimize.cpp.o -MF CMakeFiles/mcsema-lift-11.0.dir/mcsema/BC/Optimize.cpp.o.d -o CMakeFiles/mcsema-lift-11.0.dir/mcsema/BC/Optimize.cpp.o -c ../mcsema/BC/Optimize.cpp
../mcsema/BC/Optimize.cpp:1542:21: error: no matching constructor for initialization of 'llvm::LoadInst'
                new llvm::LoadInst(ptr, load->getName(), load));
                    ^              ~~~~~~~~~~~~~~~~~~~~~~~~~~
/home/ekilmer/github/cxx-common/vcpkg/installed/x64-linux-rel/include/llvm/IR/Instructions.h:190:3: note: candidate constructor not viable: requires 4 arguments, but 3 were provided
  LoadInst(Type *Ty, Value *Ptr, const Twine &NameStr,
  ^
/home/ekilmer/github/cxx-common/vcpkg/installed/x64-linux-rel/include/llvm/IR/Instructions.h:192:3: note: candidate constructor not viable: requires 4 arguments, but 3 were provided
  LoadInst(Type *Ty, Value *Ptr, const Twine &NameStr, BasicBlock *InsertAtEnd);
  ^
/home/ekilmer/github/cxx-common/vcpkg/installed/x64-linux-rel/include/llvm/IR/Instructions.h:173:7: note: candidate constructor (the implicit copy constructor) not viable: requires 1 argument, but 3 were provided
class LoadInst : public UnaryInstruction {
      ^
/home/ekilmer/github/cxx-common/vcpkg/installed/x64-linux-rel/include/llvm/IR/Instructions.h:193:3: note: candidate constructor not viable: requires 5 arguments, but 3 were provided
  LoadInst(Type *Ty, Value *Ptr, const Twine &NameStr, bool isVolatile,
  ^
/home/ekilmer/github/cxx-common/vcpkg/installed/x64-linux-rel/include/llvm/IR/Instructions.h:195:3: note: candidate constructor not viable: requires 5 arguments, but 3 were provided
  LoadInst(Type *Ty, Value *Ptr, const Twine &NameStr, bool isVolatile,
  ^
/home/ekilmer/github/cxx-common/vcpkg/installed/x64-linux-rel/include/llvm/IR/Instructions.h:197:3: note: candidate constructor not viable: requires at least 5 arguments, but 3 were provided
  LoadInst(Type *Ty, Value *Ptr, const Twine &NameStr, bool isVolatile,
  ^
/home/ekilmer/github/cxx-common/vcpkg/installed/x64-linux-rel/include/llvm/IR/Instructions.h:199:3: note: candidate constructor not viable: requires 6 arguments, but 3 were provided
  LoadInst(Type *Ty, Value *Ptr, const Twine &NameStr, bool isVolatile,
  ^
/home/ekilmer/github/cxx-common/vcpkg/installed/x64-linux-rel/include/llvm/IR/Instructions.h:201:3: note: candidate constructor not viable: requires at least 6 arguments, but 3 were provided
  LoadInst(Type *Ty, Value *Ptr, const Twine &NameStr, bool isVolatile,
  ^
/home/ekilmer/github/cxx-common/vcpkg/installed/x64-linux-rel/include/llvm/IR/Instructions.h:205:3: note: candidate constructor not viable: requires 8 arguments, but 3 were provided
  LoadInst(Type *Ty, Value *Ptr, const Twine &NameStr, bool isVolatile,
  ^
1 error generated.
[30/31] Building CXX object CMakeFiles/mcsema-lift-11.0.dir/mcsema/CFG/CFG.cpp.o
../mcsema/CFG/CFG.cpp:458:11: warning: 'SetTotalBytesLimit' is deprecated: Please use the single parameter version of SetTotalBytesLimit(). The second parameter is ignored. [-Wdeprecated-declarations]
  cstream.SetTotalBytesLimit(512 * 1024 * 1024, -1);
          ^
/home/ekilmer/github/cxx-common/vcpkg/installed/x64-linux-rel/include/google/protobuf/io/coded_stream.h:394:3: note: 'SetTotalBytesLimit' has been explicitly marked deprecated here
  PROTOBUF_DEPRECATED_MSG(
  ^
/home/ekilmer/github/cxx-common/vcpkg/installed/x64-linux-rel/include/google/protobuf/port_def.inc:171:53: note: expanded from macro 'PROTOBUF_DEPRECATED_MSG'
#define PROTOBUF_DEPRECATED_MSG(msg) __attribute__((deprecated(msg)))
                                                    ^
../mcsema/CFG/CFG.cpp:1121:39: warning: implicit conversion changes signedness: '::mcsema::ExceptionFrame_Action' to 'uint64_t' (aka 'unsigned long') [-Wsign-conversion]
      frame_var->action_index = entry.action();
                              ~ ~~~~~~^~~~~~~~
2 warnings generated.
ninja: build stopped: subcommand failed.
```

--- a/.github/workflows/ci.yml
+++ b/.github/workflows/ci.yml
@@ -25,7 +25,7 @@ jobs:
     needs: [VersionFile]
     strategy:
       matrix:
-        llvm: ["900", "1000"]
+        llvm: ["900", "1000", "1100"]
         ubuntu: ["20.04", "18.04"]
     steps:
     - uses: actions/checkout@v2

--- a/.github/workflows/vcpkg_ci.yml
+++ b/.github/workflows/vcpkg_ci.yml
@@ -18,7 +18,8 @@ jobs:
           - { name: 'ubuntu', tag: '20.04' }
         llvm: [
           '9',
-          '10'
+          '10',
+          '11'
           ]
 
     runs-on: ubuntu-20.04
@@ -76,7 +77,8 @@ jobs:
           ]
         llvm: [
           '9',
-          '10'
+          '10',
+          '11'
           ]
 
     runs-on: ${{ matrix.os }}

--- a/.github/workflows/vcpkg_ci.yml
+++ b/.github/workflows/vcpkg_ci.yml
@@ -42,7 +42,7 @@ jobs:
         shell: bash
         run: |
           # Change to master after testing
-          git clone --depth=1 --branch compat-llvm-11-mcsema https://github.com/lifting-bits/remill.git
+          git clone --depth=1 --branch master https://github.com/lifting-bits/remill.git
           cd remill
           ./scripts/build.sh --llvm-version ${{ matrix.llvm }}
           cmake --build remill-build --target install -- -j "$(nproc)"

--- a/.github/workflows/vcpkg_ci.yml
+++ b/.github/workflows/vcpkg_ci.yml
@@ -42,7 +42,7 @@ jobs:
         shell: bash
         run: |
           # Change to master after testing
-          git clone --depth=1 --branch master https://github.com/lifting-bits/remill.git
+          git clone --depth=1 --branch compat-llvm-11-mcsema https://github.com/lifting-bits/remill.git
           cd remill
           ./scripts/build.sh --llvm-version ${{ matrix.llvm }}
           cmake --build remill-build --target install -- -j "$(nproc)"

--- a/mcsema/BC/Function.cpp
+++ b/mcsema/BC/Function.cpp
@@ -46,16 +46,14 @@
 #include <remill/Arch/Instruction.h>
 #include <remill/BC/ABI.h>
 #include <remill/BC/Annotate.h>
+#include <remill/BC/Compat/CallSite.h>
+#include <remill/BC/Compat/Cloning.h>
 #include <remill/BC/Compat/Error.h>
 #include <remill/BC/Compat/ScalarTransforms.h>
 #include <remill/BC/IntrinsicTable.h>
 #include <remill/BC/Lifter.h>
 #include <remill/BC/Util.h>
 #include <remill/BC/Version.h>
-
-#if LLVM_VERSION_NUMBER < LLVM_VERSION(11, 0)
-#include <llvm/IR/CallSite.h>
-#endif
 
 #include <memory>
 #include <unordered_map>
@@ -1678,23 +1676,13 @@ static llvm::Function *LiftFunction(const NativeModule *cfg_module,
   return lifted_func;
 }
 
-#if LLVM_VERSION_NUMBER < LLVM_VERSION(11, 0)
-using Calls_t = std::vector<llvm::CallSite>;
+using Calls_t = std::vector<remill::compat::llvm::CallSite>;
 
-static bool ShouldInline(const llvm::CallSite &cs) {
-#else
-using Calls_t = std::vector<llvm::CallBase*>;
-
-static bool ShouldInline(const llvm::CallBase *cs) {
-#endif
+static bool ShouldInline(const remill::compat::llvm::CallSite &cs) {
   if (!cs) {
     return false;
   }
-#if LLVM_VERSION_NUMBER < LLVM_VERSION(11, 0)
   auto callee = cs.getCalledFunction();
-#else
-  auto callee = cs->getCalledFunction();
-#endif
   return callee &&
          remill::HasOriginType<remill::Semantics, remill::ExtWrapper>(callee);
 }
@@ -1703,11 +1691,7 @@ static Calls_t InlinableCalls(llvm::Function &func) {
   Calls_t out;
   for (auto &bb : func) {
     for (auto &inst : bb) {
-#if LLVM_VERSION_NUMBER < LLVM_VERSION(11, 0)
-      auto cs = llvm::CallSite(&inst);
-#else
-      auto cs = llvm::dyn_cast<llvm::CallBase>(&inst);
-#endif
+      auto cs = remill::compat::llvm::CallSite(&inst);
       if (ShouldInline(cs)) {
         out.push_back(std::move(cs));
       }
@@ -1718,15 +1702,9 @@ static Calls_t InlinableCalls(llvm::Function &func) {
 
 static void InlineCalls(llvm::Function &func) {
   for (auto &cs : InlinableCalls(func)) {
-#if LLVM_VERSION_NUMBER < LLVM_VERSION(11, 0)
     if (auto call = llvm::dyn_cast<llvm::CallInst>(cs.getInstruction())) {
       llvm::InlineFunctionInfo info;
       llvm::InlineFunction(call, info);
-#else
-    if (auto call = llvm::dyn_cast<llvm::CallInst>(cs)) {
-      llvm::InlineFunctionInfo info;
-      llvm::InlineFunction(*call, info);
-#endif
     }
   }
 }

--- a/mcsema/BC/Function.cpp
+++ b/mcsema/BC/Function.cpp
@@ -25,7 +25,6 @@
 #pragma clang diagnostic ignored "-Wswitch-enum"
 #include <gflags/gflags.h>
 #include <glog/logging.h>
-#include <llvm/IR/CallSite.h>
 #include <llvm/IR/Constants.h>
 #include <llvm/IR/DataLayout.h>
 #include <llvm/IR/DebugInfo.h>
@@ -53,6 +52,10 @@
 #include <remill/BC/Lifter.h>
 #include <remill/BC/Util.h>
 #include <remill/BC/Version.h>
+
+#if LLVM_VERSION_NUMBER < LLVM_VERSION(11, 0)
+#include <llvm/IR/CallSite.h>
+#endif
 
 #include <memory>
 #include <unordered_map>
@@ -211,7 +214,11 @@ static llvm::Function *GetValueTracer(void) {
     if (reg->type == gWordType && reg != pc_reg) {
       ss << reg->name << format;
       args.push_back(
+#if LLVM_VERSION_NUMBER < LLVM_VERSION(11, 0)
           new llvm::LoadInst(reg->AddressOf(state_ptr, block), "", block));
+#else
+          new llvm::LoadInst(reg->type, reg->AddressOf(state_ptr, block), "", block));
+#endif
     }
   }
   ss << '\n';
@@ -1671,13 +1678,23 @@ static llvm::Function *LiftFunction(const NativeModule *cfg_module,
   return lifted_func;
 }
 
+#if LLVM_VERSION_NUMBER < LLVM_VERSION(11, 0)
 using Calls_t = std::vector<llvm::CallSite>;
 
 static bool ShouldInline(const llvm::CallSite &cs) {
+#else
+using Calls_t = std::vector<llvm::CallBase*>;
+
+static bool ShouldInline(const llvm::CallBase *cs) {
+#endif
   if (!cs) {
     return false;
   }
+#if LLVM_VERSION_NUMBER < LLVM_VERSION(11, 0)
   auto callee = cs.getCalledFunction();
+#else
+  auto callee = cs->getCalledFunction();
+#endif
   return callee &&
          remill::HasOriginType<remill::Semantics, remill::ExtWrapper>(callee);
 }
@@ -1686,7 +1703,11 @@ static Calls_t InlinableCalls(llvm::Function &func) {
   Calls_t out;
   for (auto &bb : func) {
     for (auto &inst : bb) {
+#if LLVM_VERSION_NUMBER < LLVM_VERSION(11, 0)
       auto cs = llvm::CallSite(&inst);
+#else
+      auto cs = llvm::dyn_cast<llvm::CallBase>(&inst);
+#endif
       if (ShouldInline(cs)) {
         out.push_back(std::move(cs));
       }
@@ -1697,9 +1718,15 @@ static Calls_t InlinableCalls(llvm::Function &func) {
 
 static void InlineCalls(llvm::Function &func) {
   for (auto &cs : InlinableCalls(func)) {
+#if LLVM_VERSION_NUMBER < LLVM_VERSION(11, 0)
     if (auto call = llvm::dyn_cast<llvm::CallInst>(cs.getInstruction())) {
       llvm::InlineFunctionInfo info;
       llvm::InlineFunction(call, info);
+#else
+    if (auto call = llvm::dyn_cast<llvm::CallInst>(cs)) {
+      llvm::InlineFunctionInfo info;
+      llvm::InlineFunction(*call, info);
+#endif
     }
   }
 }

--- a/mcsema/BC/Optimize.cpp
+++ b/mcsema/BC/Optimize.cpp
@@ -1539,7 +1539,12 @@ static void GlobalizeStateStructures(void) {
                 llvm::PointerType::get(itp->getType(), 0));
             ptr = TryGetRegAlias(ptr, offset);
             itp->replaceAllUsesWith(
-                new llvm::LoadInst(ptr, load->getName(), load));
+#if LLVM_VERSION_NUMBER < LLVM_VERSION(11, 0)
+                new llvm::LoadInst(ptr, load->getName(), load)
+#else
+                new llvm::LoadInst(ptr->getType(), ptr, load->getName(), load)
+#endif
+                );
             to_remove.push_back(itp);
           }
         }

--- a/scripts/docker-lifter-entrypoint.sh
+++ b/scripts/docker-lifter-entrypoint.sh
@@ -56,6 +56,9 @@ case ${LLVM_VERSION} in
   llvm100*)
     V=10.0
   ;;
+  llvm110*)
+    V=11.0
+  ;;
   *)
     echo "Unknown LLVM version: ${LLVM_VERSION}"
     exit 1

--- a/tools/mcsema_lift/Lift.cpp
+++ b/tools/mcsema_lift/Lift.cpp
@@ -305,7 +305,11 @@ struct ABILibsLoader {
 
     for (auto &alias : abi_lib->aliases()) {
       if (auto fn = llvm::dyn_cast<llvm::Function>(alias.getAliasee())) {
+#if LLVM_VERSION_NUMBER < LLVM_VERSION(11, 0)
         CloneFunction(*fn, alias.getName());
+#else
+        CloneFunction(*fn, alias.getName().str());
+#endif
       }
     }
 

